manifest/oci_index.go

```diff
@@ -89,11 +98,20 @@ func (index *OCI1Index) ChooseInstance(ctx *types.SystemContext) (digest.Digest,
 				Variant:      d.Platform.Variant,
 			}
 			if platform.MatchesPlatform(imagePlatform, wantedPlatform) {
-				return d.Digest, nil
+				matchedDigest = d.Digest
+				// Always give priority to zstd if more than one image
```

Comment:
I am not sure about doing that unconditionally.  In theory, this can break some use cases that rely on always picking gzip.  Shouldn't there be (at least) a way to opt out of this behavior?  Could be config knob?

Reply:
That would imply walking the entire list, recording the matching ones but the pick the first ZSTD one.

Reply:
A config knob sounds good to me , let me think of a way without breaking any API.

> That would imply walking the entire list, recording the matching ones but the pick the first ZSTD one.

True

Reply:
- _Typically_ `CopySystemImage` is used for pulls. In that case we don’t need an option at all.
- Pushes don’t have a Zstd source, so they don’t need an option either.
- That leaves `skopeo copy docker://… docker://…`, which indeed uses `CopySystemImage` by default. I’m not too sure we need an option for this _now_ (as opposed to some later `copy --compression={ensure:zstd, strip:zstd, only:gzip}` mechanism) , but it wouldn’t hurt.

I’d expect a new option in `copy.Option` to be the natural way to express this.

And yes, that would require adding an `internal/manifest.List`, so that we can add/change a method, somewhat similarly to how we now have `internal/image`. This will be necessary to allow adding / removing compression variants in the future, either way.

And that `internal/manifest` work probably be a separate (fairly large) PR just to build that infrastructure, with no new features.

Reply:
FWIW the `internal/manifest` could only contain the `List` parts we need, no need to migrate everything.

Reply:
A new field in `copy.Option` sounds good.  This would nicely cover CLI use cases - however it will be exposed on the CLI.

But I also desire an option in a configuration file.  I think that registries.conf can be a nice host for that.  This could, for now, be sufficient to make the ZSTD-preference conditional (and ideally opt-in).  This way, all users of c/image would be covered (incl. CRI-O if OpenShift decided to default to ZSTD in the future).

Reply:
I have made is configurable in `systemContext` and as discussed yesterday later on we can extend `copy.Option` to propagate it to systemContext

Reply:
Resolving this so we can discuss on the new config option.